### PR TITLE
JAMES-2855 Upgrade assertj libs

### PR DESCRIPTION
**Pull request by Arsnael**
Not stable yet... At least `MailImplTest` is breaking... It seems assertj cannot compare anymore the Map field `attributes` between two `MailImpl` objects

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR] Failures: 
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.duplicateFactoryMethodShouldGenerateNewObjectWithSameValuesButName:158 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@24aed80c> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@3a52dba3> in <org.apache.james.server.core.MailImpl@2a5c8d3f>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["message", "name"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplConstructionShouldSetDefaultValuesOnUnspecifiedFields:98 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@33f676f6> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@4c5ae43b> in <org.apache.james.server.core.MailImpl@264f218>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["sender", "name", "recipients", "lastUpdated"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplConstructionWithMimeMessageShouldSetSpecifiedFields:132 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@32eff876> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@74e52303> in <org.apache.james.server.core.MailImpl@47af7f3d>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["message", "lastUpdated"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializable:321 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@5aa9e4eb> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@6989da5e> in <org.apache.james.server.core.MailImpl@385c9627>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializableWithCollectionAttribute:363 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@5167f57d> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@64b0598> in <org.apache.james.server.core.MailImpl@18f8cd79>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializableWithOptionalAttribute:342 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@54e041a4> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@2c78324b> in <org.apache.james.server.core.MailImpl@79defdc>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
```

**Comment by Arsnael** [comment]
```
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR] Failures: 
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.duplicateFactoryMethodShouldGenerateNewObjectWithSameValuesButName:158 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@24aed80c> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@3a52dba3> in <org.apache.james.server.core.MailImpl@2a5c8d3f>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["message", "name"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplConstructionShouldSetDefaultValuesOnUnspecifiedFields:98 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@33f676f6> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@4c5ae43b> in <org.apache.james.server.core.MailImpl@264f218>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["sender", "name", "recipients", "lastUpdated"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplConstructionWithMimeMessageShouldSetSpecifiedFields:132 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@32eff876> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@74e52303> in <org.apache.james.server.core.MailImpl@47af7f3d>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields but <["message", "lastUpdated"]>
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializable:321 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@5aa9e4eb> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@6989da5e> in <org.apache.james.server.core.MailImpl@385c9627>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializableWithCollectionAttribute:363 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@5167f57d> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@64b0598> in <org.apache.james.server.core.MailImpl@18f8cd79>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
[1e942becbce019488e53c10ba15161df1018b05f] [ERROR]   MailImplTest.mailImplShouldBeSerializableWithOptionalAttribute:342 
[1e942becbce019488e53c10ba15161df1018b05f] Expecting value <java.util.stream.ReferencePipeline$Head@54e041a4> in field <"attributes"> but was <java.util.stream.ReferencePipeline$Head@2c78324b> in <org.apache.james.server.core.MailImpl@79defdc>.
[1e942becbce019488e53c10ba15161df1018b05f] Comparison was performed on all fields
```
Known error. Will investigate today

File: protocols/imap/src/test/java/org/apache/james/imap/processor/base/MailboxEventAnalyserTest.java [hunk 1]
```diff
@@ -208,7 +208,9 @@ public void testShouldSetUidWhenSystemFlagChange() {
 
         testee.event(update);
 
-       assertThat(testee.flagUpdateUids().iterator()).containsExactly(UID);
+       assertThat(testee.flagUpdateUids().iterator())
```
Comment (trantienduchn) [inline protocols/imap/src/test/java/org/apache/james/imap/processor/base/MailboxEventAnalyserTest.java]: you can remove `.iterator()` then able to remove `.toIterable()`. Similar to some tests below

Comment (Arsnael) [reply protocols/imap/src/test/java/org/apache/james/imap/processor/base/MailboxEventAnalyserTest.java]: Let me check :)

**Comment by Arsnael** [comment]
```
[ERROR]   MailsShouldBeWellReceived.mailsShouldBeWellReceived:63 » ConditionTimeout Cond...
```
test this please

**Comment by Arsnael** [comment]
```
[b6b5597162823187eb52cc5615805c606b6e3c6f] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[b6b5597162823187eb52cc5615805c606b6e3c6f] LastClientMsg: c SEARCH FLAGGED
[b6b5597162823187eb52cc5615805c606b6e3c6f] Expected: '\* SEARCH 1 2 3 4'
[b6b5597162823187eb52cc5615805c606b6e3c6f] Actual   : '* SEARCH'
[b6b5597162823187eb52cc5615805c606b6e3c6f] 	at org.apache.james.mpt.imapmailbox.external.james.DeploymentValidation.validateDeployment(DeploymentValidation.java:81)
```
... test this please

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java [hunk 2]
```diff
@@ -265,7 +271,9 @@ public void messagesRetrievedUsingFetchTypeHeaderShouldHaveHeaderDataLoaded() th
         MessageMapper.FetchType fetchType = MessageMapper.FetchType.Headers;
         Iterator<MailboxMessage> retrievedMessageIterator = messageMapper.findInMailbox(benwaInboxMailbox, MessageRange.one(message1.getUid()), fetchType, LIMIT);
         assertThat(retrievedMessageIterator.next()).isEqualTo(message1, fetchType);
-        assertThat(retrievedMessageIterator).isEmpty();
+        assertThat(retrievedMessageIterator)
+            .toIterable()
```
Comment (blackheaven) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java]: why have you done that?

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 3]
```diff
@@ -1201,7 +1201,8 @@ public void reprocessingAllTaskShouldClearMailRepository() throws Exception {
             .basePath(TasksRoutes.BASE)
             .get(taskId + "/await");
 
-        assertThat(mailRepository.list()).isEmpty();
+        assertThat(mailRepository.list()).toIterable()
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: CR

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 4]
```diff
@@ -1555,7 +1563,7 @@ public void reprocessingOneTaskShouldRemoveMailFromRepository() throws Exception
             .basePath(TasksRoutes.BASE)
             .get(taskId + "/await");
 
-        assertThat(mailRepository.list())
+        assertThat(mailRepository.list()).toIterable()
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: CR please

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 5]
```diff
@@ -1695,7 +1707,7 @@ public void reprocessingOneTaskShouldNotEnqueueUnknownMailKey() throws Exception
             .basePath(TasksRoutes.BASE)
             .get(taskId + "/await");
 
-        assertThat(customQueue.browse())
+        assertThat(customQueue.browse()).toIterable()
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: CR

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java [hunk 6]
```diff
@@ -117,7 +117,8 @@ default void processedMailsShouldNotDecreaseSize() throws Exception {
     default void browseShouldReturnEmptyByDefault() throws Exception {
         ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
 
-        assertThat(items).isEmpty();
+        assertThat(items).toIterable()
```
Comment (blackheaven) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java]: CR

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java [hunk 7]
```diff
@@ -128,7 +129,8 @@ default void browseShouldReturnSingleElement() throws Exception {
 
         ManageableMailQueue.MailQueueIterator items = getManageableMailQueue().browse();
 
-        assertThat(items).extracting(ManageableMailQueue.MailQueueItemView::getMail)
+        assertThat(items).toIterable()
```
Comment (blackheaven) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java]: CR

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java (hunk 2 above)
Comment (Arsnael) [reply mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java]: Seems like it's the new syntax, had to explicitly transform it to an iterable... Or it just doesn't compile

**Pull request closed without merging**